packages/sanity/src/structure/comments/src/context/comments/CommentsProvider.tsx

```diff
@@ -50,8 +51,8 @@ export interface CommentsProviderProps {
  */
 export const CommentsProvider = memo(function CommentsProvider(props: CommentsProviderProps) {
   const {children, documentId, documentType, isCommentsOpen, onCommentsOpen} = props
+  const commentsEnabled = useCommentsEnabled()
```

Comment:
I'd like to keep the `CommentsProvider` not dependent on other contexts. I think it would be better to move this logic (in `handleSetStatus`) to the `handleChangeView` function in the `CommentsInspector` instead.

Reply:
I'm not 100% sure about this, this context it's already depending on `useCommentsSetup`, it can be moved to `commentsInspector` but if later we add another button or any other async action that sets the `resolved` states we will need to duplicate the logic added here.